### ISSUE TITLE
Newline in Mathjax broken by jupyterlab 4.0.0.

### ISSUE THREAD

**Issue by lgautier** [jupyterlab/jupyterlab#14570]
After migrating to jupyterlab 4.0.0, newlines in mathjax sections are no longer working. 

## Description

The LaTeX / mathjax syntax for newlines (`//`) no longer has an effect on the text rendered.

## Reproduce

For example, prior to 4.0.0 the following showed 2 lines. With 4.0.0 it shows all the content in one line.

```markdown
$$
2x = x + 1 \\
x = 1
$$
```

## Expected behavior

A new line when `//` is used, like it used to before jupyter 4.0.0.

## Context

- Operating System and version: Ubuntu 22.04
- Browser and version: Firefox 112.0.1
- JupyterLab version: 4.0.0

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
	/opt/software/python/py310_mini_env/bin
	/opt/adb-fastboot/platform-tools
	/usr/local/sbin
	/usr/local/bin
	/usr/sbin
	/usr/bin
	/sbin
	/bin
	/usr/games
	/usr/local/games
	/snap/bin
	/snap/bin

sys.path:
	/opt/software/python/py310_mini_env/bin
	/usr/lib/python310.zip
	/usr/lib/python3.10
	/usr/lib/python3.10/lib-dynload
	/opt/software/python/py310_mini_env/lib/python3.10/site-packages

sys.executable:
	/opt/software/python/py310_mini_env/bin/python3

sys.version:
	3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]

platform.platform():
	Linux-5.19.0-41-generic-x86_64-with-glibc2.35

which -a jupyter:
	/opt/software/python/py310_mini_env/bin/jupyter

pip list:
	Package                  Version
	------------------------ --------
	anyio                    3.6.2
	argon2-cffi              21.3.0
	argon2-cffi-bindings     21.2.0
	arrow                    1.2.3
	asttokens                2.2.1
	async-lru                2.0.2
	attrs                    23.1.0
	Babel                    2.12.1
	backcall                 0.2.0
	beautifulsoup4           4.12.2
	bleach                   6.0.0
	certifi                  2023.5.7
	cffi                     1.15.1
	charset-normalizer       3.1.0
	comm                     0.1.3
	debugpy                  1.6.7
	decorator                5.1.1
	defusedxml               0.7.1
	executing                1.2.0
	fastjsonschema           2.16.3
	fqdn                     1.5.1
	idna                     3.4
	ipykernel                6.23.1
	ipython                  8.13.2
	isoduration              20.11.0
	jedi                     0.18.2
	Jinja2                   3.1.2
	json5                    0.9.14
	jsonpointer              2.3
	jsonschema               4.17.3
	jupyter_client           8.2.0
	jupyter_core             5.3.0
	jupyter-events           0.6.3
	jupyter-lsp              2.1.0
	jupyter_server           2.5.0
	jupyter_server_terminals 0.4.4
	jupyterlab               4.0.0
	jupyterlab-pygments      0.2.2
	jupyterlab_server        2.22.1
	MarkupSafe               2.1.2
	matplotlib-inline        0.1.6
	mistune                  2.0.5
	nbclient                 0.7.4
	nbconvert                7.4.0
	nbformat                 5.8.0
	nest-asyncio             1.5.6
	notebook_shim            0.2.3
	packaging                23.1
	pandocfilters            1.5.0
	parso                    0.8.3
	pexpect                  4.8.0
	pickleshare              0.7.5
	pip                      22.0.2
	platformdirs             3.5.1
	prometheus-client        0.16.0
	prompt-toolkit           3.0.38
	psutil                   5.9.5
	ptyprocess               0.7.0
	pure-eval                0.2.2
	pycparser                2.21
	Pygments                 2.15.1
	pyrsistent               0.19.3
	python-dateutil          2.8.2
	python-json-logger       2.0.7
	PyYAML                   6.0
	pyzmq                    25.0.2
	requests                 2.30.0
	rfc3339-validator        0.1.4
	rfc3986-validator        0.1.1
	Send2Trash               1.8.2
	setuptools               59.6.0
	six                      1.16.0
	sniffio                  1.3.0
	soupsieve                2.4.1
	stack-data               0.6.2
	terminado                0.17.1
	tinycss2                 1.2.1
	tomli                    2.0.1
	tornado                  6.3.2
	traitlets                5.9.0
	typing_extensions        4.5.0
	uri-template             1.2.0
	urllib3                  2.0.2
	wcwidth                  0.2.6
	webcolors                1.13
	webencodings             0.5.1
	websocket-client         1.5.1
</pre>
</details>

**Comment by krassowski** [comment]
This is a known issue of MathJax 3 we [hesitated for a long time](https://github.com/jupyterlab/jupyterlab/issues/7218#issuecomment-1404705339) on whether to proceed with upgrade or wait for MathJax 4 which fixes this (but is still [in development](https://github.com/mathjax/MathJax-src/commits/develop)). The old MathJax 2 renderer will be soon available to install as a drop-in replacement for users who rely on this feature.

**Comment by JasonWeill** [comment]
Related MathJax 3 issue: https://github.com/mathjax/MathJax/issues/2312

**Comment by krassowski** [comment]
[`jupyterlab-mathjax2`](https://pypi.org/project/jupyterlab-mathjax2/) package is now available on PyPI (thanks @fcollonval)

**Comment by lgautier** [comment]
Thanks!

**Comment by krassowski** [comment]
MathJax v4.0 which fixes the new line issue is now officially released:
- https://docs.mathjax.org/en/latest/upgrading/whats-new-4.0.html
- https://github.com/mathjax/MathJax/releases/tag/4.0.0
- https://docs.mathjax.org/en/latest/upgrading/whats-new-4.0/breaking.html

We could now consider upgrading, or adding `jupyterlab-mathjax4` package as an interim solution.
